sympy/printing/str.py

```diff
@@ -74,13 +74,16 @@ def _print_BooleanTrue(self, expr):
     def _print_BooleanFalse(self, expr):
         return "False"
 
+    def _print_Not(self, expr):
+        return '~(%s)' %(expr.args[0])
+
     def _print_And(self, expr):
-        return '%s(%s)' % (expr.func, ', '.join(sorted(self._print(a) for a in
-            expr.args)))
+        return '%s' %(' & '.join([self.parenthesize(arg, 38)
```

Comment:
I don't think it's a bug.

In python, precedence of `and` (That is, Boolean and) is less than precedence of relational operators.
However, precedence of `&` (That is, bitwise &) is greater than relational operators.

As `And` objects in SymPy represent Boolean `and` operation, `PRECEDENCE["And"]` should be less than `PRECEDENCE["Relational"]`.
However, we want to represent `And` objects using `&` operator. Therefore, I've used a number.